### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by victor-paumier**
- Fix preRemove buddy on edit profile and improve event priorities
- Allow admin to match
- Fix nationalities stats when more than 26 different values
- Fix null UserNotification
- Improve sorting of languages on register & profile pages
- Improve popup list buddies on matching summary
- Fix circular references in listener

### CODE REVIEW

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -154,26 +155,33 @@ public function showReport(User $user): Response
...
      */
-    public function showBuddiesList(UserRepository $userRepo, User $user): Response
+    public function showBuddiesList(BuddyRepository $buddyRepo, User $user): Response
     {
         $this->denyAccessUnlessGranted(UserVoterEnum::SHOW, $user);
 
         return $this->render('Buddy/list.html.twig', [
             'user' => $user,
-            'buddies' => $userRepo->getCurrentBuddies($user, $user->isLocal()),
+            'buddies' => $buddyRepo->getBuddiesByUser(
```
Comment (JeremieSamson) [inline src/Controller/UserController.php]: here you passed $user and $user->isLocal() in the function arguments, the less the better, IMO it's a better good practice to pass less argument and do the logic inside the function. 
Not sure the boolean is from another var in another place ?

File: src/EventListener/SectionListener.php [hunk 2]
```diff
@@ -18,22 +18,11 @@
...
-        HobbyRepository $hobbyRepo,
-        MotivationRepository $motivationRepo,
-        SemesterRepository $semesterRepo,
-        MailManager $mailManager,
-        SemesterHelper $semesterHelper
-    ) {
-        $this->hobbyRepo = $hobbyRepo;
-        $this->motivationRepo = $motivationRepo;
-        $this->semesterRepo = $semesterRepo;
+    public function __construct(MailManager $mailManager, SemesterHelper $semesterHelper)
```
Comment (JeremieSamson) [inline src/EventListener/SectionListener.php]: 👍

File: templates/Dashboard/Stats/bc_view_content.html.twig [hunk 3]
```diff
@@ -157,7 +157,7 @@
                                 <div>
                                     <span class="chart-legend__color {{ 'chart-legend__pie--' ~ range('a','z')[loop.index0%10] }}"></span>
                                     <p class="insight__legend">{{ university }}</p>
-                                    <p class="pl-4">{{ 'dashboard.insights.universities.nb_users'|trans ~' '~ universities.series[loop.index0] }}</p>
+                                    <p class="pl-4">{{ 'dashboard.insights.legend.nb_users '|trans ~' '~ universities.series[loop.index0] }}</p>
```
Comment (JeremieSamson) [inline templates/Dashboard/Stats/bc_view_content.html.twig]:
```suggestion
                                    <p class="pl-4">{{ 'dashboard.insights.legend.nb_users'|trans ~' '~ universities.series[loop.index0] }}</p>
```

File: src/Controller/UserController.php (hunk 1 above)
Comment (victor-paumier) [reply src/Controller/UserController.php]: The problem here is that sometime I need this boolean to be different than the value of $user->isLocal(). See [BuddyListener](https://github.com/ESNFranceG33kTeam/sf_buddysystem/blob/4c26fd35665a84ccbff18d5b63adc07652f99e59/src/EventListener/BuddyListener.php#L92)

Comment (JeremieSamson) [reply src/Controller/UserController.php]: ok got it

**Pull request merged**